src/fsharp/FSharp.Core/mailbox.fs

```diff
@@ -93,7 +52,7 @@ namespace Microsoft.FSharp.Control
     [<AutoSerializable(false)>]
     type Mailbox<'Msg>(cancellationSupported: bool) =
         let mutable inboxStore  = null
-        let mutable arrivals = new Queue<'Msg>()
+        let arrivals = Queue<'Msg>()
         let syncRoot = arrivals
```

Comment:
Probably sync root and arrivals are now the exact same thing and one can be removed